streams/sha3/sha3_stream.cc

```diff
@@ -1,18 +1,19 @@
 #include "sha3_stream.h"
+#include "eacirc/streams.h"
 #include "sha3_factory.h"
 #include "sha3_interface.h"
 #include <algorithm>
 
 template <typename I>
 static void
-hash_data(sha3_interface& hasher, const I& data, std::uint8_t* hash, std::size_t hash_size) {
+hash_data(sha3_interface& hasher, const I& data, std::uint8_t* hash, const std::size_t hash_size) {
     using std::to_string;
 
-    int status = hasher.Init(int(8 * hash_size));
+    int status = hasher.Init(int(hash_size * 8));
     if (status != 0)
         throw std::runtime_error("cannot initialize hash (code: " + to_string(status) + ")");
 
-    status = hasher.Update(data.data(), 8 * data.size());
+    status = hasher.Update(&(*data.begin()), 8 * (data.end() - data.begin()));
```

Comment:
Very supportive, now I really know, how to solve this! I spent with this an hour to figure out I do not know how to do it better. Today again 15 minutes, I did not found better solution (I deleted this and after 15 minutes, I came again with this :/). So?